### DISCUSSION TITLE
How to install binary package from GitHub repository?

### DISCUSSION THREAD

**Discussion by kevinmatthes** [vlang/v#18837] [Questions and Answers] [answered]
In order to learn V practically, I began to work on a small CLI project whose Git repository is hosted on GitHub.  I would like to install the latest repository state as a binary executable on my local machine to test the software.  The project itself has no dependencies (except for the V standard library) and consists only of the `main` module.

When I try install my software with `v install --git`, followed by the URL to the repository, the code base itself is downloaded and saved within `~/.vmodules/` but the binary is not compiled.  I know that I oversaw something but even after working through the language documentation and the source code of the installation logic, I still have no idea how to have the binary project compiled on installation.

Any ideas?

**Comment by JalonSolov** [comment, marked as answer]
`v install` is for installing modules that will be pulled in and compiled with other code.  It isn't meant to install and build standalone code.

If your code needs something more complex than `v .` to build, we recommend including a `build.vsh` file with your code.  The user will still need to run the `build.vsh` file themselves after pulling down your code - there is no automated system for this.

The other alternative would be to create an install package for the OS, but you'll likely have to create a separate package for each OS variation, as well as every different package manager a user might use on that OS.
